### PR TITLE
wait for group cache to avoid flake in templateinstance test

### LINKED ISSUE #16775: [Conformance][templates] templateinstance security tests should pass security tests
https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/16765/test_pull_request_origin_extended_conformance_gce/9386/

may (or may not) be similar to #16654

### PR DESCRIPTION
**Pull request by jim-minter**
fixes #16775 (hopefully).

@smarterclayton @bparees @enj

### CODE REVIEW

File: test/extended/templates/templateinstance_security.go [hunk 1]
```diff
@@ -67,180 +68,216 @@ var _ = g.Describe("[Conformance][templates] templateinstance security tests", f
...
 	)
 
-	g.BeforeEach(func() {
-		adminuser = createUser(cli, "adminuser", bootstrappolicy.AdminRoleName)
-		edituser = createUser(cli, "edituser", bootstrappolicy.EditRoleName)
-		editgroup = createGroup(cli, "editgroup", bootstrappolicy.EditRoleName)
-		editbygroupuser = createUser(cli, "editbygroupuser", "")
-		addUserToGroup(cli, editbygroupuser.Name, editgroup.Name)
-	})
+	g.Context("", func() {
```
Comment (enj) [inline test/extended/templates/templateinstance_security.go]: Not sure what an empty string here means.

Comment (jim-minter) [reply test/extended/templates/templateinstance_security.go]: it's an idiom that devex is currently using in tests to ensure that the inner AfterEach runs before k8s' global AfterEach destroys the namespace

Comment (enj) [reply test/extended/templates/templateinstance_security.go]: Sounds crazy.  But I guess that is ginko...

**Comment by enj** [comment]
/approve

I will leave final LGTM to @bparees

**Comment by bparees** [comment]
@jim-minter go ahead and mark this as fixing/closing https://github.com/openshift/origin/issues/16775

if it recurs after this merges we'll reopen/open a new flake.

**Pull request merged**